src/Http/Exception/HttpException.php

```diff
@@ -30,4 +30,44 @@ class HttpException extends Exception
      * @inheritDoc
      */
     protected $_defaultCode = 500;
+
+    /**
+     * @var array
+     */
+    protected $headers = [];
+
+    /**
+     * Sets http response header.
+     *
+     * @param string $header Header name
+     * @param string|array|null $value Header value
+     * @return void
+     */
+    public function setHeader(string $header, $value = null): void
+    {
+        $this->setHeaders([$header => $value]);
+    }
+
+    /**
+     * Sets an array of http response headers.
+     *
+     * Will replace any headers with the same name.
+     *
+     * @param array $headers Array of header name and value pairs.
+     * @return void
+     */
+    public function setHeaders(array $headers): void
+    {
+        $this->headers = $headers + $this->headers;
```

Comment:
From a coding conventions POV, should a "set" function always overwrite the entire array?

Reply:
In the case where you're taking an array of things `set` operations should overwrite existing data.

Reply:
Updated. I updated the tests to test both methods properly. I also split up the array values so we can test that they are properly joined by ','.